contractsSol5/KyberHintHandler.sol

```diff
@@ -346,22 +177,139 @@ contract KyberHintHandler is IKyberHint, Utils4 {
             ethToTokenType,
             ethToTokenAddresses,
             ethToTokenSplits,
-            failedIndex
+            error
         ) = parseHintT2T(hint);
 
-        if (failedIndex == 0) {
-            tokenToEthReserveIds = new bytes8[](tokenToEthAddresses.length);
-            ethToTokenReserveIds = new bytes8[](ethToTokenAddresses.length);
+        if (error != HintErrors.NoError) throwHintError(error);
+
+        tokenToEthReserveIds = new bytes8[](tokenToEthAddresses.length);
+        ethToTokenReserveIds = new bytes8[](ethToTokenAddresses.length);
 
-            for (uint i = 0; i < tokenToEthAddresses.length; i++) {
-                tokenToEthReserveIds[i] = convertAddressToReserveId(address(tokenToEthAddresses[i]));
+        for (uint i = 0; i < tokenToEthAddresses.length; i++) {
+            tokenToEthReserveIds[i] = convertAddressToReserveId(address(tokenToEthAddresses[i]));
+        }
+        for (uint i = 0; i < ethToTokenAddresses.length; i++) {
+            ethToTokenReserveIds[i] = convertAddressToReserveId(address(ethToTokenAddresses[i]));
+        }
+    }
+
+    function parseHint(bytes memory hint)
+        internal
+        view
+        returns(
+            TradeType tradeType,
+            IKyberReserve[] memory addresses,
+            uint[] memory splits,
+            HintErrors verified
+        )
+    {
+        bytes8[] memory reserveIds;
+
+        (tradeType, reserveIds, splits) = abi.decode(hint, (TradeType, bytes8[], uint[]));
+        verified = verifyData(tradeType, reserveIds, splits);
+
+        if (verified == HintErrors.NoError) {
+            addresses = new IKyberReserve[](reserveIds.length);
+            
+            for (uint i = 0; i < reserveIds.length; i++) {
+                addresses[i] = IKyberReserve(convertReserveIdToAddress(reserveIds[i]));
+            }
+        } else {
+            splits = new uint[](0);
+        }
+    }
+
+    function parseHintT2T(bytes memory hint)
+        internal
+        view
+        returns(
+            TradeType t2eType,
+            IKyberReserve[] memory t2eAddresses,
+            uint[] memory t2eSplits,
+            TradeType e2tType,
+            IKyberReserve[] memory e2tAddresses,
+            uint[] memory e2tSplits,
+            HintErrors verified
```

Comment:
valid seems to me a bit more suitable.
not sure though

Reply:
Hmmm okay. Using Valid is fine too.